src/utils/command_handlers.py

```diff
@@ -0,0 +1,65 @@
+from managers import ContactManager
+from models import Contact
+
+def handle_add_contact(manager: ContactManager) -> None:
+    """
+    Handles the addition of a new contact to the contact manager.
+
+    Prompts the user to enter details for the new contact, such as name, address, phone number, email, and birthday.
+    It performs basic validation to ensure that the name is not empty and attempts to add the contact to the manager.
+    If an error occurs during the process, it prints an appropriate error message.
+
+    Parameters:
+        manager (ContactManager): An instance of ContactManager to manage contacts.
+    """
+    try:
+        name = input("Enter name: ").strip()
+        if not name:
+            print("Name cannot be empty.")
+            return
+        
+        address = input("Enter address: ").strip()
+        phone_number = input("Enter phone number: ").strip()
+        email = input("Enter email: ").strip()
+        birthday = input("Enter birthday (DD-MM-YYYY): ").strip()
+
+        new_contact = Contact(name=name, address=address, phone_number=phone_number, email=email, birthday=birthday)
+        manager.add_contact(new_contact)
+    except Exception as ex:
+        print(f"An error occurred while adding the contact: {ex}")
+
+
+def handle_search_contact(manager: ContactManager) -> None:
+    """
+    Handles the search for contacts based on the specified search type.
+
+    Prompts the user to select a search type ('name', 'email', or 'phone') and enter the search query.
+    It then performs the search using the appropriate method from the ContactManager and displays the results.
+    If an error occurs during the search, it prints an appropriate error message.
+
+    Parameters:
+        manager (ContactManager): An instance of ContactManager to manage contacts.
+    """
+    search_type = input("Search by (name/email/phone): ").strip().lower()
+    query = input("Enter the search query: ").strip()
+
+    try:
```

Comment:
Do you think this makes sense or is it an unnecessary complexity in the code?
search_map = {
    "name": manager.search_by_name, 
    "email": manager.search_by_email, 
    "phone": manager.search_by_phone_number}

if search_type in search_map:
    results = search_map.get(search_type)(query)
else:
    print("Invalid search type. Please choose 'name', 'email', or 'phone'.")
    return

Reply:
Good idea, I forgot about this life hack.